app/controllers/users_controller.rb

```diff
@@ -1,8 +1,16 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, except: %i(show new create)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: %i(destroy)
+
   def new
     @user = User.new
   end
 
+  def index
+    @users = User.all.page(params[:page]).per Settings.per_page
+  end
+
   def show
     @user = User.find_by id: params[:id]
```

Comment:
Nếu id không tồn tại em có vào trang show được không? ví dụ params[:id] = 10000 hoặc -1 chẳng hạn

Reply:
hiện tại thì không ạ, vì trang show em đang cho in ra user.name

Reply:
@user = User.find_by id: params[:id] em thấy hàm này dùng chung nhiều k? Hãy viết 1 hàm load_user rồi dùng before_action để sử dụng hàm này cho các action em mong muốn giống before_action :correct_user, only: %i(edit update), nhớ check đk k tìm thấy user và có tìm thấy?

Reply:
vâng